src/shapes/textbox.class.js

```diff
@@ -78,6 +86,33 @@
      */
     splitByGrapheme: false,
 
+    /**
+     * While editing handle differently
+     * @private
+     * @param {string} key
+     * @param {*} value
+     */
+    _set: function (key, value) {
+      if (key === 'width') {
+        if (typeof this.maxWidth === 'number' && this.maxWidth < value) {
+          value = this.maxWidth;
+        }
+        if (typeof this.minWidth === 'number' && this.minWidth > value) {
+          value = this.minWidth;
+        }
```

Comment:
also infinity enables this:
```suggestion
value = Math.max(this.minWidth, Math.min(value, this.maxWidth))
```